akka-http-core/src/main/scala/akka/http/model/MediaType.scala

```diff
@@ -253,33 +274,34 @@ object MediaTypes extends ObjectRegistry[(String, String), MediaType] {
 
   def forExtension(ext: String): Option[MediaType] = extensionMap.get(ext.toRootLowerCase)
 
-  private def app(subType: String, compressible: Boolean, binary: Boolean, fileExtensions: String*) = register {
-    new NonMultipartMediaType("application", subType, compressible, binary, immutable.Seq(fileExtensions: _*)) {
+  private def app(subType: String, compressible: Boolean, encoding: Encoding, fileExtensions: String*) = register {
+    new NonMultipartMediaType("application", subType, compressible, encoding, immutable.Seq(fileExtensions: _*)) {
       override def isApplication = true
     }
   }
   private def aud(subType: String, compressible: Boolean, fileExtensions: String*) = register {
-    new NonMultipartMediaType("audio", subType, compressible, binary = true, immutable.Seq(fileExtensions: _*)) {
+    new NonMultipartMediaType("audio", subType, compressible, encoding = Encoding.Binary, immutable.Seq(fileExtensions: _*)) {
       override def isAudio = true
     }
   }
-  private def img(subType: String, compressible: Boolean, binary: Boolean, fileExtensions: String*) = register {
-    new NonMultipartMediaType("image", subType, compressible, binary, immutable.Seq(fileExtensions: _*)) {
+  private def img(subType: String, compressible: Boolean, encoding: Encoding, fileExtensions: String*) = register {
+    new NonMultipartMediaType("image", subType, compressible, encoding, immutable.Seq(fileExtensions: _*)) {
       override def isImage = true
     }
   }
   private def msg(subType: String, fileExtensions: String*) = register {
-    new NonMultipartMediaType("message", subType, compressible = true, binary = false, immutable.Seq(fileExtensions: _*)) {
+    new NonMultipartMediaType("message", subType, compressible = true, encoding = Encoding.Fixed(HttpCharsets.`UTF-8`),
```

Comment:
Hmm, I'm not so sure about this. From RFC 2046:

>    No encoding other than "7bit", "8bit", or "binary" is permitted for
   the body of a "message/rfc822" entity.  The message header fields are
   always US-ASCII in any case, and data within the body can still be
   encoded, in which case the Content-Transfer-Encoding header field in
   the encapsulated message will reflect this.

In any case, it seems there's no charset parameter defined for any of those media types, so maybe binary would be the most general to use here.